source/secure_tunneling.c

```diff
@@ -450,9 +451,9 @@ static int s_secure_tunneling_send_data(struct aws_secure_tunnel *secure_tunnel,
                 return AWS_OP_ERR;
             }
         }
-        aws_mutex_lock(&mutex);
-        aws_condition_variable_wait(&condition_variable, &mutex);
-        aws_mutex_unlock(&mutex);
+        aws_mutex_lock(&secure_tunnel->send_data_mutex);
+        aws_condition_variable_wait(&secure_tunnel->send_data_condition_variable, &secure_tunnel->send_data_mutex);
```

Comment:
Only non-trivial concern remaining is wrt the condition (or lack of) here.  It is possible for the notify to happen-before the wait can get invoked.  Unlike events, there's no memory of the previous notice.  Even flipping a bool back and forth would be sufficient state.